crates/wallet/src/wallet/mod.rs

```diff
@@ -1833,9 +1832,28 @@ impl Wallet {
         psbt: &mut Psbt,
         sign_options: SignOptions,
     ) -> Result<bool, SignerError> {
+        let tx = &psbt.unsigned_tx;
         let chain_tip = self.chain.tip().block_id();
+        let prev_txids = tx
+            .input
+            .iter()
+            .map(|txin| txin.previous_output.txid)
+            .collect::<HashSet<Txid>>();
+        let confirmation_heights = self
+            .indexed_graph
+            .graph()
+            .list_canonical_txs(&self.chain, chain_tip)
+            .filter(|canon_tx| prev_txids.contains(&canon_tx.tx_node.txid))
+            .take(prev_txids.len())
```

Comment:
It's for a small performance gain. We may not return anything since `filter` filters it out. However, the `CanonicalIter` will continue to try canonicalize. Will add a comment.

> If I make both prev_txids and confirmation_heights empty, no test in wallet fails. It seems like some execution paths are not covered by tests, so it is hard to tell whether logic of the implementation changed or not.

Lets create a ticket since this is out of scope for this PR.

Reply:
I decided to include a comment on #642 instead.